### PR TITLE
Add KeepActivityPrivate condition to showPrivate

### PR DESCRIPTION
**Pull request by tyroneyeh**
For user public activity page is empty issue

### CODE REVIEW

File: routers/web/user/profile.go [hunk 1]
```diff
@@ -124,7 +124,7 @@ func Profile(ctx *context.Context) {
 		}
 	}
 
-	showPrivate := ctx.IsSigned && (ctx.Doer.IsAdmin || ctx.Doer.ID == ctx.ContextUser.ID)
+	showPrivate := ctx.IsSigned && (ctx.Doer.IsAdmin || ctx.Doer.ID == ctx.ContextUser.ID || !ctx.Doer.KeepActivityPrivate)
```
Comment (lunny) [inline routers/web/user/profile.go]: User himself or admin could still visit those information.

Comment (tyroneyeh) [reply routers/web/user/profile.go]: yes

Comment (wxiaoguang) [reply routers/web/user/profile.go]: Are you sure the Doer is the correct user? Doer means the visitor.

Comment (tyroneyeh) [reply routers/web/user/profile.go]: Oh sorry, corrected

**Comment by wxiaoguang** [comment]
I do not think this change is right and I haven't understood the problem.

For the new code: if KeepActivityPrivate=true, then any user can see the context user's private orgs? I do not think it's right to expose private orgs to any visitor.

**Comment by tyroneyeh** [comment]
This modification should be on the public activity tab, not part of the organization. It happens to use the same variable and has been moved to the public activity

Thanks

**Comment by wxiaoguang** [comment]
I am still not sure whether it is right. For example, context user has private activities in its private org and KeepActivityPrivate=false, should a visitor user see the activities in the private org? Is the private org exposed to the visitor by the activity feeds?

**Comment by tyroneyeh** [comment]
The problem is that personal public activity can only be viewed by yourself and the system administrator. It should be open to others to view, or you can set it as private

**Comment by wxiaoguang** [comment]
Is the private org exposed to the visitor by the activity feeds?

Does the org owner want and allow to do so?

**Comment by tyroneyeh** [comment]
Or where can I judge that only the same organization can browse public activity?

**Comment by wxiaoguang** [comment]
No idea... I don't have the knowledge at the moment. It doesn't seem to be an easy problem IMO.

**Pull request closed without merging**